Info message when user selects a tier2 language

leivaburto opened the pull request:
Adding an informative message when user selects a tier2 language (community-based translation).

SebastienGllmt commented:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/2608559/57566034-bd86de80-7402-11e9-8786-26f823a2611b.png">

Maybe we should add a background for the one on the wallet creation page? It's kind of hard to read depending on your screen size

SebastienGllmt commented:
For thanking the contributors, can you do the following:

Create a new i18n message that says
> "Thank you to the following for their contribution: "
Then create a second i18n message that is simply blank

Now when a new contributor joins our translation system, they can add their own name without having to change the code.

In our code, check if the 2nd message is blank.
If not, display the 1st + 2nd message

SebastienGllmt commented on app/components/settings/categories/GeneralSettings.js:
```diff
@@ -10,13 +10,29 @@ import LocalizableError from '../../../i18n/LocalizableError';
     defaultMessage: '!!!Language',
   },
-
+  languageSelectLabelInfo: {
```
Maybe better to make these global messages instead of duplicating them? (see `global-messages.js`)

You could even add a helper method in there that constructs the `contributors` string so you don't have to duplicate this logic either

SebastienGllmt commented on app/config/languagesConfig.js:
```diff
@@ -0,0 +1,5 @@
+
+export const tier1Languages = ['en-US', 'ja-JP', 'ru-RU', 'es-ES'];
+
+// In the future we should add name of non-tier1 translation contributors in here
```
remove

SebastienGllmt commented on app/i18n/locales/zh-Hant.json:
```diff
@@ -122,6 +122,10 @@
   "settings.general.aboutYoroi.website": "Yoroi網站",
   "settings.general.aboutYoroi.youtube": "EMURGO YouTube",
   "settings.general.languageSelect.label": "語言",
+  "settings.general.languageSelect.labelInfo": "The selected language translation is fully provided by the community",
```
FYI you should only ever modify the English file in a PR. Otherwise it may cause merge conflicts with the translation system we use.

leivaburto replied:
Good to know! Thanks

SebastienGllmt commented on app/stores/toplevel/ProfileStore.js:
```diff
@@ -302,7 +302,7 @@ export default class ProfileStore extends Store {
 
   _redirectToRoot = () => {
-    this.actions.router.goToRoute.trigger({ route: ROUTES.WALLETS.ROOT });
+    this.actions.router.goToRoute.trigger({ route: ROUTES.WALLETS.ADD });
```
Why was this changed? Doesn't match the function name anymore.

leivaburto replied:
ProfileStore reactions were triggering some automatic redirections and I thought this would solve it, but it didn't. I changed this back to the original and I added a new action for updating locale and redirect to terms of use.

SebastienGllmt commented:
Is it possible to add a back arrow on the Terms of Use page like we have in dialogs? I'm don't mind approving this PR without it but it may be useful

<img width="453" alt="image" src="https://user-images.githubusercontent.com/2608559/57829316-7ffec880-77e9-11e9-9620-bc0995bfcdfc.png">

leivaburto commented:
I added a new action and changed a bit the conditions on reactions. I also had to update current tests to include the new behavior (which requires clicking on "Continue" button at the Language Selection Form). All checks are passing now.

sarveshwar-singh commented:
@leivaburto 

- `LanguageSelectionForm.js` + `LanguageSelectionForm.scss`
![image](https://user-images.githubusercontent.com/19986226/58013265-40114b80-7b31-11e9-86b3-069e8ee70295.png)
![image](https://user-images.githubusercontent.com/19986226/58013314-58816600-7b31-11e9-9de2-6e71771b1e29.png)

- `GeneralSettings.js` + `GeneralSettings.scss`
![image](https://user-images.githubusercontent.com/19986226/58013406-836bba00-7b31-11e9-8bc4-79a55e72d4ab.png)
![image](https://user-images.githubusercontent.com/19986226/58013436-91213f80-7b31-11e9-9de5-f74763d2d583.png)

Seems to be similar, how about making a separate common component?

sarveshwar-singh commented on app/i18n/global-messages.js:
```diff
@@ -249,3 +270,12 @@ export const environmentSpecificMessages = {
   }),
 };
+
+export function listOfTranslators(contributorsList, contributorsAck): string {
```
@leivaburto 

- if you make a separate component them this whole function `listOfTranslators` can be moved to that component, don't feel this function should be here.

- `if (contributorsList !== globalMessages.translationContributors.defaultMessage)`
comparing to `defaultMessage` will get you in trouble if we decide to use language specific names (that still need to decided), but we never use `defaultMessage` for logical things.
I feel, the whole if condition is not needed here, `if (contributorsList !== globalMessages.translationContributors.defaultMessage)`

leivaburto replied:
Regarding your second comment, the idea is this: there would be an empty string, and contributors who are willing to do so will put their names in. Empty string messages fallback to defaultMessage value, so contributorsList would take that value. What do you mean by using language specific names? AFAIK strings would fallback to default values defined in global messages, regardless of language, which in this case is `!!!List of contributors`. Or is there another strategy to be considered?

leivaburto replied:
Regarding your first comment, it makes sense. Actually, I was working on PR #494 and I had to make something similar, so I'm planning to use that same component in here (it's called InformativeMessage). The component is generic, so I'm trying to figure out where would be better to put the method you mention. Anyway, I want to make sure the component I added makes sense (I think it does). Could you take a look at it? and if everything is OK I could use it here. Btw, thanks for your feedback.

sarveshwar-singh replied:
@leivaburto 
> What do you mean by using language specific names?
Got clear after your explanation.

sarveshwar-singh commented:
@leivaburto 
![image](https://user-images.githubusercontent.com/19986226/58014769-85834800-7b34-11e9-8f6e-1873f9dae68f.png)

Plz try to fix those, let me know if you need some help.
